sender-stackdriver/src/main/java/zipkin2/reporter/stackdriver/internal/AwaitableUnaryClientCallListener.java

```diff
@@ -28,16 +29,20 @@
 
   Object result; // guarded by this
 
+  long timeoutMs = 5000; // how long to wait for server response in milliseconds
+
   /**
    * Blocks until {@link #onClose}. Throws if no value was received, multiple
-   * values were received, or there was a status error.
+   * values were received, there was a status error, or waited longer than {@link #timeoutMs}.
    */
   V await() throws IOException {
     boolean interrupted = false;
     try {
       while (true) {
         try {
-          countDown.await();
+          if (!countDown.await(this.timeoutMs, TimeUnit.MILLISECONDS)) {
+            throw new IllegalStateException("Send failed after waiting for " + timeoutMs + "ms.");
```

Comment:
"timeout waiting for onClose. timeoutMs=" + timeoutMs + ", valueSet="+valueSet

This will introduce maintenance as maybe there's a weird bug where the value is set, yet we fail anyway (because some bug resulted in onClose never being called). If we find this then there's no point failing I suppose, but we should find out first